### PR TITLE
Fixe opposite selection dialog

### PR DESCRIPTION
**Pull request by azoitl**

### CODE REVIEW

File: plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/widgets/OppositeSelectionDialog.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+		}
+
+		private void closePopup() {
+			setPopupClosable(true);
+			dialogArea.getShell().close();
+		}
+
+		@Override
+		public void keyPressed(final KeyEvent e) {
+			if (e.character == SWT.CR || e.character == SWT.ESC) {
```
Comment (MandKastner) [inline plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/widgets/OppositeSelectionDialog.java:226]: somehow the keylistener is not always triggered when pressing escape.. do you know how to fix that?

**Review by MandKastner — APPROVED** [review]

File: plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/widgets/OppositeSelectionDialog.java (hunk 1 above)
Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/widgets/OppositeSelectionDialog.java:226]: I had the impression that this was the case before my merge commit. 

could it be that in your regaining the focus code you first need to get the focus for your shell, before you give it to the table viewer?

Can we investigate that in a separate PR?

Comment (MandKastner) [reply plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/widgets/OppositeSelectionDialog.java:226]: sorry i forgot something.. please just add setPopupClosable(true) in the focusGained() of the Focus listener

Comment (azoitl) [reply plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/widgets/OppositeSelectionDialog.java:226]: Done

**Review by MandKastner — APPROVED** [review]

**Pull request merged**